src/pretalx/schedule/models/slot.py

```diff
@@ -27,9 +27,10 @@ class TalkSlot(LogMixin, models.Model):
     is_visible = models.BooleanField()
     start = models.DateTimeField(null=True)
     end = models.DateTimeField(null=True)
+    slot_index = models.PositiveIntegerField(default=0)
 
     class Meta:
-        unique_together = (('submission', 'schedule'),)
+        unique_together = (('submission', 'schedule', 'slot_index'),)
 
```

Comment:
>Why did you add this field?

`slot_index` is used for unique identification of the TalkSlot - as `submission` + `schedule` is not unique anymore...
i use this to create / update and delete if `slot_count` changes in `submission/models/submission.py` at `def update_talk_slots()`

Reply:
Why do we need unique identification of a TalkSlot? I'm not sure this additional field is worth the overhead. We can just count the existing slots for create/update/delete cycles.

Reply:
hm - then we could not track if a slot was moved for the changelog generation.
i will try and think a bit about it...

Reply:
@rixx i tried this but did not come to a result :-(
how can i  do the 
```python
for index in range(self.slot_count):
    TalkSlot.objects.update_or_create(
        submission=self,
        schedule=self.event.wip_schedule,
        defaults={'is_visible': is_visible},
    )
```
then?
as fare as i understand the `unique_together = (('submission', 'schedule'),)`
means there is only one TalkSlot object with these pair.
so my for loop just updates this one obj multiple times...
(thats also the result i see :-) )

so i need one additional key for the `unique_together` to have multiple Slots with the same submission and schedule i think. 
this does not have to be a self-defined field - it would be enough if this is auto-generated.
i have reworked the update/delete section so that it is independent off `slot_index`.

but at this point i don't know / question myself if you had a complete different idea in mind how this could work...

Reply:
Yes, you would need to remove the ``unique_together`` constraint from that model – but why not do that instead of adding yet another indexed key?

Reply:
done!